app/controllers/users_controller.rb

```diff
@@ -16,7 +15,25 @@ def create
     end
   end
 
-  def show
+  def edit
+    if current_admin? && User.find(params[:id]) == current_user
+      @user = current_user
+    else
+      render file: 'app/views/errors/not_found.html.erb'
+    end
+  end
+
+  def update
+    @user = current_user
+    # byebug
+    if @user.authenticate(params[:current_password])
```

Comment:
Not positive. That does seem weird. Is there any chance that `.authenticate` is doing more than we think? I notice that in your form you have a `current_password_confirmation` line. Maybe take that out and change the name in the form from `current_password` to `password`? Believe you should be able to leave the `new_password` and `new_password_confirmation`, but maybe look into that as a next step?